src/VirtoCommerce.Platform.Core/Extensions/TypeExtensions.cs

```diff
@@ -67,5 +68,74 @@ private static string PrettyPrintRecursive(Type type, int depth)
                 : $"{nameParts[0]}<{string.Join(",", genericArguments.Select(t => PrettyPrintRecursive(t, depth + 1)))}>";
         }
 
+
+        /// <summary>
+        /// Iterates recursively over each public property of object to gather member values.
+        /// </summary>
+        /// <param name="original"></param>
+        /// <returns></returns>
+        public static IEnumerable<KeyValuePair<string, object>> ObjectWalker(this object original)
```

Comment:
I would like to suggest renaming to `TraverseObjectGraph` because the method name must contain a verb.